ai/cohere/v0/component_test.go

```diff
@@ -0,0 +1,180 @@
+package cohere
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	cohereSDK "github.com/cohere-ai/cohere-go/v2"
+	qt "github.com/frankban/quicktest"
+	"github.com/instill-ai/component/base"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	apiKey = "cohere-api-key"
+)
+
+func TestComponent_Execute(t *testing.T) {
+	c := qt.New(t)
+
+	bc := base.Component{Logger: zap.NewNop()}
+	connector := Init(bc)
+
+	c.Run("ok - supported task", func(c *qt.C) {
+		task := textGenerationTask
+
+		_, err := connector.CreateExecution(nil, nil, task)
+		c.Check(err, qt.IsNil)
+	})
+
+	c.Run("nok - unsupported task", func(c *qt.C) {
+		task := "FOOBAR"
+
+		_, err := connector.CreateExecution(nil, nil, task)
+		c.Check(err, qt.ErrorMatches, "unsupported task")
+	})
+
+}
+
+func TestComponent_Generation(t *testing.T) {
```

Comment:
Question
`func TestComponent_Generation(t *testing.T) {`

Do embed & rerank also belong to Generation?
<img width="407" alt="image" src="https://github.com/instill-ai/component/assets/64747455/85cb3c25-e8bb-48b5-8be1-acbdd07eb744">

Reply:
Nope. The embedding and rerank models are probably some sort of sentence transformer to map the text meaning to a finite vector space, thus not generative model. link: https://sbert.net/
Fixed naming.